jetbrains-core/src/software/aws/toolkits/jetbrains/core/executables/ExecutableManager.kt

```diff
@@ -102,17 +107,31 @@ class DefaultExecutableManager : PersistentStateComponent<List<ExecutableState>>
 
     private fun updateInternalState(type: ExecutableType<*>, instance: ExecutableInstance) {
         val resolved = instance as? ExecutableWithPath
-        val newPersistedState = ExecutableState(type.id,
+        val newPersistedState = ExecutableState(
+            type.id,
             resolved?.executablePath?.toString(),
             resolved?.autoResolved,
-            resolved?.executablePath?.lastModifiedOrNull()?.toMillis())
-        internalState[type.id] = Triple(newPersistedState, instance, resolved?.executablePath?.lastModified())
+            resolved?.executablePath?.lastModifiedOrNull()?.toMillis()
+        )
+        val lastModified = try {
+            resolved?.executablePath?.lastModified()
```

Comment:
When do we see this error surface?

Reply:
when we call lastModified on an invalid path

Reply:
ahhh